nixos/modules/config/swap.nix

```diff
@@ -5,6 +5,12 @@ with lib;
 
 let
 
+  # From modules/tasks/filesystems.nix
+  addCheckDesc = desc: elemType: check: types.addCheck elemType check
+  // { description = "${elemType.description} (with check: ${desc})"; };
+  isNonEmpty = s: (builtins.match "[ \t\n]*" s) == null;
+  nonEmptyStr = addCheckDesc "non-empty" types.str isNonEmpty;
```

Comment:
Could you add `nonEmptyStr` to `lib/types.nix`? Please keep the `description` simple, like `"non-empty string"`. There's no need for it to be verbose and mechanical like that. Doing so also removes the need for `addCheckDesc`.